### ISSUE TITLE
Python-level Linear Modularization.

### ISSUE THREAD

**Issue by Artiprocher** [mit-han-lab/nunchaku#24]
Hello, I am the developer of [DiffSynth-Studio](https://github.com/modelscope/DiffSynth-Studio), another diffusion engine. We are very surprised by the amazing speedup achieved by SVDQuant. We hope to bring this technology to more tech enthusiasts.

We noticed that many model structures in the Nunchaku engine are written in C++, which makes it very difficult for us to integrate them into other projects. Based on our understanding of the SVDQuant paper, the key acceleration happens in the Linear layer. Could you provide a pre-compiled quantized Linear layer module? This would make it easier for developers to integrate it with existing technologies at the Python level.

We look forward to your reply. If you can provide such a module, we are willing to promote it to more open-source projects.

Zhongjie Duan
DiffSynth Team, ModelScope
Best regards.

**Comment by lmxyy** [comment]
Of course. We are still cleaning and improving the codebase and modularization is in our TODO list. I will let you know when getting it done.
